controllers/drplacementcontrol.go

```diff
@@ -741,7 +741,7 @@ func (d *DRPCInstance) RunRelocate() (bool, error) {
 		return !done, fmt.Errorf(errMsg)
 	}
 
-	if !d.validatePeerReady() {
+	if d.getLastDRState() != rmn.Relocating && !d.validatePeerReady() {
```

Comment:
Is this required because we set peer ready to false earlier, or additional change? If another change, can you separate it to another commit and explain the reason for this change?

Reply:
Reconcile will loop over `Relocate` multiple times as various states change (e.g current primary transitions to secondary etc.). As the PeerReady is set to false for the current generation of the spec, once relocation has started, a future loop will error out here. This additional check hence is required to pass this gate if we checked everything was fine with the prior generation and started relocation.